### ISSUE TITLE
goo.gl URL shortener support

### ISSUE THREAD

**Issue by vanushah** [cezarsa/silver_bird#81]
It would be great to be able to shorten URL with Google's new URL shortener

**Comment by darkhearth** [comment]
To use goo.gl as the default URL shortener is a must, I mean, we are using chrome after all =S

**Comment by cezarsa** [comment]
First of all, thanks a lot for using Chromed Bird, and I hope you'll understand why I'm not going to add support to goo.gl shortener right now.

If you access http://goo.gl you'll read the following notice: "Google URL Shortener is currently available for Google products and not for broader consumer use."
I know a lot of people already uses goo.gl in their products even if they aren't affiliated with Google in any way, but I'm not one of these people.

The only means one can make use of goo.gl is by using some ugly hacks including fake security tokens and fake user-agents. I don't intend to implement (or directly copy from Google toolbar, like many did) all these hacks just to support yet another shortener, and just because it's trending and it's from Google. There's already plenty of shortener options available and many of them are pretty fast and reliable.

I'm leaving this issue open because when (and if) Google decides to publish a public API for their shortener I'll be happy to add support to it, but not before.

**Comment by rsvargas** [comment]
Hi Cezar, if you want to add it here is the code:

```
GooglShortener = {
  shorten: function(longUrl, useAcct, login, apiKey, callback) {
    this.Auth = function(f){function k(){for(var c=0,b=0;b<arguments.length;b++)c=c+arguments[b]&4294967295;return c}function m(c){c=c=String(c>0?c:c+4294967296);var b;b=c;for(var d=0,i=false,j=b.length-1;j>=0;--j){var g=Number(b.charAt(j));if(i){g*=2;d+=Math.floor(g/10)+g%10}else d+=g;i=!i}b=b=d%10;d=0;if(b!=0){d=10-b;if(c.length%2==1){if(d%2==1)d+=9;d/=2}}b=String(d);b+=c;return b}function n(c){for(var b=5381,d=0;d<c.length;d++)b=k(b<<5,b,c.charCodeAt(d));return b}function o(c){for(var b=0,d=0;d<c.length;d++)b=k(c.charCodeAt(d),b<<6,b<<16,-b);return b}f={byteArray_:f,charCodeAt:function(c){return this.byteArray_[c]}};f.length=f.byteArray_.length;var e=n(f.byteArray_);e=e>>2&1073741823;e=e>>4&67108800|e&63;e=e>>4&4193280|e&1023;e=e>>4&245760|e&16383;var l="7";f=o(f.byteArray_);var h=(e>>2&15)<<4|f&15;h|=(e>>6&15)<<12|(f>>8&15)<<8;h|=(e>>10&15)<<20|(f>>16&15)<<16;h|=(e>>14&15)<<28|(f>>24&15)<<24;l+=m(h);return l};
    var url = 'http://goo.gl/api/url';
    var params = {
      user: 'toolbar@google.com',
      url: longUrl,
      auth_token: this.Auth(longUrl)
    };
    $.ajax({
      url: url,
      type: 'post',
      data: params,
      dataType: 'json',
      success: function(data, status) {
        callback(0, data.short_url);
      },
      error: function (request, status, error) {
        callback(-1, 'AJAX request failed (bad connection?)');
      },
      timeout: 6000
    });
  }
}



SHORTENERS_BACKEND = {
  /* ... */
  googl: {
    desc: 'goo.gl',
    baseUrl: '',
    backend: GooglShortener
  }
}
```

and it's available here http://github.com/rsvargas/chromed_bird

**Comment by vanushah** [comment]
Now when http://goo.gl URL shortener became available to broad audience would you add support for it?

**Comment by rsvargas** [comment]
@vanushah http://goo.gl support is available since version 1.8.0
